ui/src/routes/devices.$id.settings.tsx

```diff
@@ -30,7 +30,7 @@ export default function SettingsRoute() {
   const scrollContainerRef = useRef<HTMLDivElement>(null);
   const [showLeftGradient, setShowLeftGradient] = useState(false);
   const [showRightGradient, setShowRightGradient] = useState(false);
-  const { width } = useResizeObserver({ ref: scrollContainerRef });
+  const { width = 0 } = useResizeObserver({ ref: scrollContainerRef as React.RefObject<HTMLDivElement> });
```

Comment:
react 19.x added Ref<T | null> so this we need to explicitly cast now

Also, best to avoid width possibly being null